openvdb/doc/dependencies.txt

```diff
@@ -0,0 +1,174 @@
+/**
+
+@page dependencies Dependencies
+
+@section depContents Contents
+- @ref depComponents
+  - @ref depDependencyTable
+  - @ref depKnownIssues
+- @ref depInstallingDependencies
+  - @ref depUsingAptGet
+  - @ref depUsingHomebrew
+  - @ref depManInstall
+<!-- - @ref depContents -->
+
+------------------------------------------------------------------------------
+
+@b Note: The dependency information on this page details a complete
+requirements list for building OpenVDB as a @b standalone package. If you wish
+to use OpenVDB with a supported 3rd party DCC (Houdini/Maya), make sure you've
+first read the [Building OpenVDB](@ref build) information.
+
+------------------------------------------------------------------------------
+
+There are a number of optional and required dependencies necessary when
+building different components of OpenVDB. Below you'll find a full list of all
+dependencies, any minimum/recommended version requirements and some guides for
+getting these dependencies installed. You'll also find information on any
+known issues with particular software. Please report any issues not listed
+here to the OpenVDB TSC.
+
+Note that OpenVDB is production tested against the [VFX Reference Platform]
+(https://www.vfxplatform.com/). Not all dependencies exist in the VFX
+Reference Platform, but for those that do, their specified versions are
+@b always recommended.
+
+@section depComponents OpenVDB Components
+
+Component               | Requirements                                                                         | Optional
+----------------------- | ------------------------------------------------------------------------------------ | --------
+OpenVDB Core Library    | CMake, C++11 compiler, IlmBase::Half, TBB::tbb, ZLIB, Boost::system, Boost::iostream | Blosc, Log4cplus, OpenEXR::IlmImf
+OpenVDB Print           | Core Library dependencies                                                            | -
+OpenVDB LOD             | Core Library dependencies                                                            | -
+OpenVDB Render          | Core Library dependencies, OpenEXR, IlmBase                                          | -
+OpenVDB View            | Core Library dependencies, OpenGL, GLFW3, Boost::thread                              | -
+OpenVDB Python          | Core Library dependencies, Python, Boost::python                                     | NumPy
+OpenVDB UnitTests       | Core Library dependencies, CppUnit                                                   | -
+OpenVDB Documentation   | Doxygen                                                                              | -
+
+@subsection depDependencyTable Dependency Table
+
+Package        | Minimum | Recommended | Description                                                       | apt-get | Homebrew | Source
```

Comment:
Using boost as an example, cy2017 is 1.61 and cy2019 is 1.66. If the library builds under an earlier version of boost (i.e 1.55), should we be enforcing explicit errors if boost is < 1.61? The minimum compatible version is kind of what I wanted this column to represent and it should match the cmake behaviour we introduce. i.e. what should cmake do when:
 - A software version x isn't compatible, we error (obvious)
 - x < minimum vfx platform, but vdb builds ??
 - x > current vfx platform, but vdb builds ??

The minimum vfx platform version will always be recommended (in this case 1.61), but I'm wondering if the minimum version should just represent what is required to build openvdb, where as the recommended is the current vfx platform (and thus tested) set of versions

Reply:
> Using boost as an example, cy2017 is 1.61 and cy2019 is 1.66. If the library builds under an earlier version of boost (i.e 1.55), should we be enforcing explicit errors if boost is < 1.61? 

Yeah, I think we should. We're explicitly not testing a Boost version earlier than 1.61 so there is no guarantee that the library will work as expected. I think the best approach is to ensure that the library errors if you use dependencies that are too old, even if that particular version _might_ still work.

> i.e. what should cmake do when:
> 
> * A software version x isn't compatible, we error (obvious)
> * x < minimum vfx platform, but vdb builds ??

Error

> * x > current vfx platform, but vdb builds ??

Nothing

> 
> The minimum vfx platform version will always be recommended (in this case 1.61), but I'm wondering if the minimum version should just represent what is required to build openvdb, where as the recommended is the current vfx platform (and thus tested) set of versions

Continuing to use Boost as an example - in my opinion, the CI should test 1.61, 1.66 and the latest (currently ~1.70). We provide a strong guarantee that the library will build correctly for our supported VFX Reference Platform years and a loose guarantee for more recent versions than that. (Blosc remains an exception to this rule).

However, I do appreciate that sometimes it may be desirable to use an older dependency and take on the risk of doing so. I suggest we add support for a variable such as 'ENABLE_OLDER_DEPENDENCIES' which would downgrade the aforementioned error into a warning for all dependencies (and the error message should mention it too). That means that people are explicitly opting in to using an older dependency. We're not relying on them spotting a warning in the CMake output to avoid sinking a bunch of time into track down why they're getting segfaults due to an outdated dependency that we're no longer testing against.

Reply:
> I suggest we add support for a variable such as 'ENABLE_OLDER_DEPENDENCIES' which would downgrade the aforementioned error into a warning for all dependencies (and the error message should mention it too). 

I like this idea, I'll do that. There's no CMake system for find_package to just promote a warning instead of an error, so I'm just going to provide an advanced options which suppressed all version checks for now - otherwise I'd need to add custom logic to each call of `find_package()` or wrap it in some kind of macro

Reply:
Done - opted for `DISABLE_DEPENDENCY_VERSION_CHECKS` which defaults of OFF